src/rapidgzip/gzip/GzipReader.hpp

```diff
@@ -384,6 +530,11 @@ class GzipReader :
     std::optional<size_t> m_offsetInLastBuffers;
 
     CRC32Calculator m_crc32Calculator;
+
+    bool m_indexIsImported{ false };
+    std::shared_ptr<BlockMap> m_blockMap;
+    std::shared_ptr<WindowMap> m_windowMap{ std::make_shared<WindowMap>() };
```

Comment:
If possible, make both of these `std::shared_ptr<const ...>`, then simply copying the `shared_ptr`, as you do in your `clone` method, would be thread-safe. I think, the window map should also be created during index import. That seems semantically more correct and may avoid issues in case `importIndex` is called more than once. Although, normally, even if called more than once, I would expect it to be called with identical input data and therefore should still be idempotent.